java-squid/src/main/java/org/sonar/java/resolve/JavaSymbol.java

```diff
@@ -505,29 +504,48 @@ public MethodJavaSymbol overriddenSymbol() {
         return null;
       }
       TypeJavaSymbol enclosingClass = enclosingClass();
+      boolean unknownFound = false;
       for (JavaType.ClassJavaType superType : enclosingClass.superTypes()) {
         MethodJavaSymbol overridden = overriddenSymbolFrom(superType);
         if (overridden != null) {
-          return overridden;
+          if (!overridden.isUnknown()) {
+            return overridden;
+          } else {
+            unknownFound = true;
+          }
         }
       }
+      if (unknownFound) {
+        return Symbols.unknownMethodSymbol;
+      }
       return null;
     }
 
     @Nullable
     private MethodJavaSymbol overriddenSymbolFrom(JavaType.ClassJavaType classType) {
       if (classType.isTagged(JavaType.UNKNOWN)) {
-        return null;
+        return Symbols.unknownMethodSymbol;
       }
+      boolean unknownFound = false;
       List<JavaSymbol> symbols = classType.getSymbol().members().lookup(name);
       for (JavaSymbol overrideSymbol : symbols) {
         if (overrideSymbol.isKind(JavaSymbol.MTH)) {
           MethodJavaSymbol methodJavaSymbol = (MethodJavaSymbol) overrideSymbol;
-          if (canOverride(methodJavaSymbol) && Boolean.TRUE.equals(isOverriding(methodJavaSymbol, classType))) {
-            return methodJavaSymbol;
+          if (canOverride(methodJavaSymbol)) {
+            Boolean overriding = isOverriding(methodJavaSymbol, classType);
+            if (overriding == null) {
+              if (!unknownFound) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS134)